libmamba/src/api/configuration.cpp

```diff
@@ -25,6 +25,113 @@
 
 #include "termcolor/termcolor.hpp"
 
+static std::string
+expandvars(std::string s)
+{
+    auto res = std::vector<std::string>();
+
+    while (s.size())
+    {
+        const char c = s[0];
+#ifdef _WIN32
+        bool win_percent = c == '%';
+        if (c == '\'')
+        {
+            // No expansion within single quotes (Windows only)
+            auto j = s.find("'", 1);
+            if (j == std::string::npos)
+            {
+                res.push_back(s);
+                break;
+            }
+            else
+            {
+                res.push_back(s.substr(0, j + 1));
+                s = s.substr(j + 1);
+            }
+        }
+        else
+#else
+        bool win_percent = false;
+#endif
+            if (c == '$' || win_percent)
+        {
+            // Variable or $$/%%
+            auto has_next = s.size() > 1;
+            if (has_next && s[1] == (win_percent ? '%' : '$'))
+            {
+                // $$/%%
+                res.push_back(s.substr(1, 1));
+                s = s.substr(2);
+            }
+            else if (has_next && (win_percent ? isalnum(s[1]) : s[1] == '{'))
+            {
+                // ${var} or %var%
+                auto j = win_percent ? s.find("%", 1) : s.find("}", 2);
+                if (j == std::string::npos)
+                {
+                    res.push_back(s);
+                    break;
+                }
+                else
+                {
+                    auto var = s.substr(win_percent ? 1 : 2, j - (win_percent ? 1 : 2));
+                    auto val = std::getenv(var.c_str());
```

Comment:
I think it might be nice to do a preprocessing before calling `std::getenv` to make sure that `var` only contains ASCII chars and no weird things like space or `"`. If that is the case we can just return the un-modified version.

Reply:
Note that Posix supports variables using non-ASCII chars, and as a consequence so do `os.path.expandsvars` and Conda. Should not be something that's widely used though.

Reply:
Would you prefer throwing an error on non-ASCII chars or dealing with them the way we deal with unset vars? (= Keep the original string)